examples/generative/opt2_text_generation_with_distribution.py

```diff
@@ -0,0 +1,174 @@
+"""
+Title: OPT2 Text Generation with KerasNLP and Keras Distribution API
+Author: Qianli (Scott) Zhu
+Date created: 01/11/2024
+Last modified: 01/11/2024
+Description: Use OPT2 model and Keras distribution API to do text generation.
+Accelerator: GPU
+"""
+
+"""
+In this tutorial, you will learn to use [KerasNLP](https://keras.io/keras_nlp/)
+to load a pre-trained Large Language Model (LLM) - [OPT-2 model](https://arxiv.org/abs/2205.01068)
+(originally invented by Meta), finetune and generate with a distribute hardware
+setting.
+"""
+
+"""
+##  Before we begin
+
+Colab offers different kinds of runtimes. Make sure to go to **Runtime ->
+Change runtime type** and choose the GPU Hardware Accelerator runtime
+(which should have >12G host RAM and ~16G GPU RAM) since you will finetune the
+OPT-2 model. Running this tutorial on CPU runtime will take hours.
+
+Also note that this example was originally created with 8 V100 GPUs, explicitly
+to simulate how to do inference of large model with limited hardwares.
+"""
+
+"""
+## Install KerasNLP, Choose Backend and Import Dependencies
+
+This examples uses the latest distribution API from [Keras](https://keras.io/keras/).
+The API is currently supporting JAX backend, and we are adding more backends
+support in the coming future.
+"""
+
+import os
+
+# This will allow JAX to scale more to fully leverage all the available GPU memory.
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+
+import jax
+
+# We have 8 V100 GPUs, and each of which has 16G of GPU memory.
+# It will not be enough memory on a single device to host all the model weights
+# and optimizer state.
+# We are going to show case how to distribute the large model weights, so that
+# a popular LLM model (7B param) can be finetuned on a previous generation of
+# hardware.
+print(jax.devices())
+
+os.environ['KERAS_BACKEND'] = "jax"
+
+import keras
+print(keras.version())
+print(keras.backend.backend())
+
+keras.mixed_precision.set_global_policy("mixed_float16")
+
+import keras_nlp
+
+
+"""
+## Introduction to KerasNLP
```

Comment:
I suggest focusing the example purely on the distribution aspects, so we can replace the KerasNLP intro with ~1 sentence. Meanwhile maybe we could flesh out the distribution part, e.g. include fine-tuning or other inference performance considerations.